runelite-client/src/test/java/net/runelite/client/plugins/grounditems/GroundItemsPluginTest.java

```diff
@@ -0,0 +1,239 @@
+	}
+
+	@Test
+	public void shouldNotRecolorHiddenMenuEntry()
+	{
+		String itemName = "Shovel";
+		int itemQuantity = 1;
+		int itemGEPrice = 10;
+		int itemItemPrice = 10;
+		Color hiddenColor = Color.GRAY;
+
+		MenuEntry menuEntry = prepareMenuEntryTest(itemName, itemQuantity, itemGEPrice, itemItemPrice, itemName, "", Color.PINK, hiddenColor);
+		when(config.recolorMenuHiddenItems()).thenReturn(true);
+
+		groundItemsPlugin.onItemSpawned(new ItemSpawned(tile, tileItem));
+		groundItemsPlugin.onMenuEntryAdded(new MenuEntryAdded(menuEntry.getOption(), menuEntry.getTarget(), menuEntry.getType(), menuEntry.getIdentifier(), menuEntry.getParam0(), menuEntry.getParam1()));
+
+		Assert.assertEquals(String.format("%s%s", ColorUtil.colorTag(hiddenColor), itemName), menuEntry.getTarget());
+	}
+
+	private MenuEntry prepareMenuEntryTest(String itemName, Integer itemQuantity, Integer itemGEPrice, Integer itemItemPrice, String hiddenItemsConfig, String highlightItemsConfig, Color highlightedColor, Color hiddenColor)
+	{
+		MenuEntry menuEntry = new MenuEntry();
+
+		int x = 0;
+		int y = 0;
+		int plane = 0;
+		int itemId = 2000;
+		WorldPoint worldPoint = new WorldPoint(x, y, plane);
+
+		menuEntry.setOption("Take");
+		menuEntry.setTarget(itemName);
+		menuEntry.setIdentifier(itemId);
+		menuEntry.setType(MenuAction.GROUND_ITEM_THIRD_OPTION.getId());
+		menuEntry.setParam0(x);
+		menuEntry.setParam1(y);
+
+		when(config.getHiddenItems()).thenReturn(hiddenItemsConfig);
+		when(config.getHighlightItems()).thenReturn(highlightItemsConfig);
+
+		when(config.getHighlightOverValue()).thenReturn(0);
+		when(config.insaneValuePrice()).thenReturn(10_000_000);
+		when(config.highValuePrice()).thenReturn(1_000_000);
+		when(config.mediumValuePrice()).thenReturn(100_000);
+		when(config.lowValuePrice()).thenReturn(1_000);
+
+		when(config.highlightedColor()).thenReturn(highlightedColor);
+		when(config.insaneValueColor()).thenReturn(Color.RED);
+		when(config.highValueColor()).thenReturn(Color.ORANGE);
+		when(config.mediumValueColor()).thenReturn(Color.YELLOW);
+		when(config.lowValueColor()).thenReturn(Color.WHITE);
+		when(config.hiddenColor()).thenReturn(hiddenColor);
+
+		lenient().when(config.valueCalculationMode()).thenReturn(ValueCalculationMode.HIGHEST);
+		when(config.menuHighlightMode()).thenReturn(MenuHighlightMode.BOTH);
+
+		when(tileItem.getId()).thenReturn(itemId);
+		when(tile.getItemLayer()).thenReturn(itemLayer);
+		when(itemManager.getItemComposition(itemId)).thenReturn(itemComposition);
+		when(client.getLocalPlayer()).thenReturn(player);
+		when(player.getWorldLocation()).thenReturn(worldPoint);
+		when(tile.getWorldLocation()).thenReturn(worldPoint);
+		when(itemComposition.getNote()).thenReturn(-1);
+		when(client.getMenuEntries()).thenReturn(new MenuEntry[]{menuEntry});
+		when(client.getScene()).thenReturn(scene);
+		when(scene.getTiles()).thenReturn(new Tile[][][]{new Tile[][]{new Tile[]{tile}}});
```

Comment:
Instead of mocking all of this code, you can just remove it from the plugin, since it is unused

Reply:
I'm not sure what you mean. The 4 lines your highlighted are non-lenient stubs, if they aren't called during execution an error would be thrown. I assure you those lines are needed.

Reply:
If you look at the final commit I pushed, it removed the code which used this from the plugin, which makes it unnecessary.